runtime/builtin_types.go

```diff
@@ -631,15 +631,23 @@ func builtinRound(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 			return nil, raised
 		}
 	}
-	// TODO: implement this.
-	if ndigits != 0 {
-		return nil, f.RaiseType(NotImplementedErrorType, "round with ndigits is not implemented in grumpy")
-	}
 	number, isFloat := floatCoerce(args[0])
 	if !isFloat {
 		return nil, f.RaiseType(TypeErrorType, "a float is required")
 	}
-	return NewFloat(math.Floor(number + 0.5)).ToObject(), nil
+
+	if math.IsNaN(number) || math.IsInf(number, 0) || number == 0.0 {
+		return NewFloat(number).ToObject(), nil
+	}
+
+	var pow float64 = 1
+	var result float64
+	for i := 0; i < ndigits; i++ {
+		pow *= 10
+	}
+
+	result = float64(math.Floor(number*pow+0.5)) / pow
```

Comment:
Is the float64(...) necessary here? The return value of Floor should already by float64.